src/template.rs

```diff
@@ -141,13 +170,90 @@ pub mod django_rusty_templates {
-            todo!()
+            py: Python<'py>,
+            loader: Bound<'_, PyAny>,
+            encoding: &'static Encoding,
+        ) -> PyResult<Loader> {
+            if let Ok(loader_str) = loader.downcast::<PyString>() {
+                return Self::map_loader(py, loader_str.to_str()?, None, encoding);
+            }
+
+            let (loader_path, args) = unpack(loader.clone()).map_err(|e| {
+                ImproperlyConfigured::new_err(format!(
+                    "Invalid template loader: {loader}. {}",
+                    e.value(py),
+                ))
+            })?;
+
+            Self::map_loader(py, loader_path.to_str()?, Some(args), encoding)
+        }
+
+        fn map_loader(
+            py: Python<'_>,
+            loader_path: &str,
+            args: Option<Bound<'_, PyAny>>,
+            encoding: &'static Encoding,
+        ) -> PyResult<Loader> {
+            match loader_path {
+                "django.template.loaders.filesystem.Loader" => {
+                    let paths = args
+                        .map(|arg| {
+                            arg.try_iter()?
+                                .map(|item| {
+                                    let s = item?.extract::<String>()?;
+                                    Ok(PathBuf::from(s))
+                                })
+                                .collect::<PyResult<Vec<_>>>()
+                        })
+                        .transpose()?
+                        .unwrap_or_default();
+
+                    Ok(Loader::FileSystem(FileSystemLoader::new(paths, encoding)))
+                }
+                "django.template.loaders.app_directories.Loader" => {
+                    Ok(Loader::AppDirs(AppDirsLoader::new(encoding)))
+                }
+                "django.template.loaders.locmem.Loader" => {
+                    let templates = args
+                        .map(|arg| arg.extract())
+                        .transpose()?
+                        .unwrap_or_default();
+
+                    Ok(Loader::LocMem(LocMemLoader::new(templates)))
+                }
+                "django.template.loaders.cached.Loader" => {
+                    let nested_loaders = args
+                        .ok_or_else(|| {
+                            ImproperlyConfigured::new_err(
+                                "django.template.loaders.cached.Loader requires a list/tuple of loaders"
+                            )
+                        })?
+                        .try_iter()?
+                        .map(|inner_loader| Self::find_template_loader(py, inner_loader?, encoding))
+                        .collect::<PyResult<Vec<_>>>()?;
+
+                    Ok(Loader::Cached(CachedLoader::new(nested_loaders)))
+                }
+                unknown => Err(ImproperlyConfigured::new_err(format!(
+                    "Invalid template loader class: {}",
+                    unknown
+                ))),
```

Comment:
I think we will want to return an `ExternalLoader` here, but that hasn't been fully implemented yet.